### PR TITLE
ignore experiment if id recorded in meta data does not match

### PR DESCRIPTION
**Pull request by mparkhe**
`ToDo`: unit tests

### CODE REVIEW

**Review by smurching — COMMENTED** [review]
Just a couple of small comments, otherwise LGTM!

File: mlflow/utils/validation.py [hunk 1]
```diff
@@ -59,3 +59,12 @@ def _validate_run_id(run_id):
     """Check that `run_id` is a valid run ID and raise an exception if it isn't."""
     if _RUN_ID_REGEX.match(run_id) is None:
         raise MlflowException("Invalid run ID: '%s'" % run_id, error_code=INVALID_PARAMETER_VALUE)
+
+
+def _validate_experiment_id(exp_id):
+    """Check that `experiment_id`is a valid integer and rause an expcepton if it isn't."""
```
[the remark this answers is not in the document]
Comment (mparkhe) [reply mlflow/utils/validation.py]: **Whoa** @ that...

File: mlflow/store/file_store.py [hunk 2]
```diff
@@ -204,13 +213,17 @@ def get_experiment_by_name(self, name):
         return None
 
     def delete_experiment(self, experiment_id):
+        if not self._has_experiment(experiment_id):
```
[the remark this answers is not in the document]
Comment (mparkhe) [reply mlflow/store/file_store.py]: You are right. The objective here was to catch for corrupted experiment meta data. But

1. This `_has_experiment` call is not doing it
2. Now that I think about it, it prolly does not make sense to check on `delete` and `restore`.

File: mlflow/store/file_store.py [hunk 3]
```diff
@@ -185,7 +189,12 @@ def _get_experiment(self, experiment_id, view_type=ViewType.ALL):
             meta['lifecycle_stage'] = Experiment.DELETED_LIFECYCLE
         else:
             meta['lifecycle_stage'] = Experiment.ACTIVE_LIFECYCLE
-        return Experiment.from_dictionary(meta)
+        experiment = Experiment.from_dictionary(meta)
+        if int(experiment_id) != experiment.experiment_id:
+            logging.warning("Experiment ID mismatch for exp %s. ID recorded as '%s' in meta data.",
```
[the remark this answers is not in the document]
Comment (mparkhe) [reply mlflow/store/file_store.py]: Done. This is used on both `LIST`, `GET`, and `RENAME`. Change the logged warning. List will ignore this and the other 2 will raise error.

**Review by smurching — APPROVED** [review]
Thanks for the updates, LGTM!

**Pull request merged**